snuba/datasets/metrics_processor.py

```diff
@@ -37,14 +45,36 @@ def process_message(
             "org_id": message["org_id"],
             "project_id": message["project_id"],
             "metric_id": message["metric_id"],
-            "metric_type": "set",
             "timestamp": timestamp,
             "tags.key": keys,
             "tags.value": values,
-            "set_values": message["value"],
+            **self._process_values(message),
             "materialization_version": 0,
             "retention_days": message["retention_days"],
             "partition": metadata.partition,
             "offset": metadata.offset,
         }
         return InsertBatch([processed], None)
+
+
+class SetsMetricsProcessor(MetricsProcessor):
+    def _should_process(self, message: Mapping[str, Any]) -> bool:
+        return message["type"] is not None and message["type"] == "s"
+
+    def _process_values(self, message: Mapping[str, Any]) -> Mapping[str, Any]:
+        values = message["value"]
+        for v in values:
+            assert isinstance(v, int), "Illegal value in set. Int expected: {v}"
```

Comment:
Should this log an error and quit instead of dying?

Reply:
At this point (the entire dataset is in dev mode, the consumer is not even deployed in prod), I would rather catching errors early. A warning when running locally would be ignored.